app/models/auth_source.rb

```diff
@@ -17,6 +17,18 @@
 
 class AuthSource < ActiveRecord::Base
   include Authorizable
+
+  class AuthSourceNameValidator < ActiveModel::Validator
+    def validate(record)
+      if record.class != AuthSourceInternal and record.class != AuthSourceHidden
+        if record.name.eql? "Internal" || record.name.eql? "Hidden"
```

Comment:
can't have that, this does not compile. if there's something after a method call's arguments, the arguments need to be inside brackets.
this should be
```ruby
if record.name.eql?("Internal") || record.name.eql?("Hidden")
```

or better yet
```ruby
%w{Internal Hidden}.include?(record.name)
```